docs/configuration.md

```diff
@@ -3152,3 +3152,109 @@ The stage level scheduling feature allows users to specify task and executor res
+
+<p> <b> Currently push-based shuffle is only supported for Spark on YARN with external shuffle service. </b></p>
+
+### Shuffle server side configuration options
+
+<table class="table">
+<tr><th>Property Name</th><th>Default</th><th>Meaning</th><th>Since Version</th></tr>
+<tr>
+  <td><code>spark.shuffle.push.server.mergedShuffleFileManagerImpl</code></td>
+  <td>
+    <code>org.apache.spark.network.shuffle.<br />NoOpMergedShuffleFileManager</code>
+  </td>
+  <td>
+    Class name of the implementation of MergedShuffleFileManager that manages push-based shuffle. This acts as a server side config to disable or enable push-based shuffle. By default, push-based shuffle is disabled at the server side. <p> To enable push-based shuffle on the server side, set this config to <code>org.apache.spark.network.shuffle.RemoteBlockPushResolver</code></p>
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.push.server.minChunkSizeInMergedShuffleFile</code></td>
+  <td><code>2m</code></td>
+  <td>
+    <p> The minimum size of a chunk when dividing a merged shuffle file into multiple chunks during push-based shuffle. A merged shuffle file consists of multiple small shuffle blocks. Fetching the complete merged shuffle file in a single disk I/O increases the memory requirements for both the clients and the external shuffle service. Instead, the shuffle service serves the merged file in <code>MB-sized chunks</code>.<br /> This configuration controls how big a chunk can get. A corresponding index file for each merged shuffle file will be generated indicating chunk boundaries. </p>
+    <p> Setting this too high would increase the memory requirements on both the clients and the external shuffle service. </p>
+    <p> Setting this too low would increase the overall number of RPC requests to external shuffle service unnecessarily.</p>
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.push.server.mergedIndexCacheSize</code></td>
+  <td><code>100m</code></td>
+  <td>
+    The size of cache in memory which is used in push-based shuffle for storing merged index files. This cache is in addition to the one configured via <code>spark.shuffle.service.index.cache.size</code>.
+  </td>
+  <td>3.2.0</td>
+</tr>
+</table>
+
+### Client side configuration options
+
+<table class="table">
+<tr><th>Property Name</th><th>Default</th><th>Meaning</th><th>Since Version</th></tr>
+<tr>
+  <td><code>spark.shuffle.push.enabled</code></td>
+  <td><code>false</code></td>
+  <td>
+    Set to 'true' to enable push-based shuffle on the client side and works in conjunction with the server side flag spark.shuffle.server.mergedShuffleFileManagerImpl.
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.push.merge.finalize.timeout</code></td>
```

Comment:
@Ngone51 2 reasons:
1. It can be slightly confusing to configure both `finalize.timeout` as well as `results.timeout` for users.
2. Also `results.timeout` is the upper bound, if the merge results arrive sooner than the upper bound (10s default) then driver just proceeds further scheduling the other stages. But that is not the case with `finalize.timeout` which is a fixed wait time before `finalize` is called on all the external shuffle services. This would also eventually become an upper bound once we address SPARK-33701. Hope that clarifies it a bit.
Let me know your thoughts. Thanks!

Reply:
> It can be slightly confusing to configure both finalize.timeout as well as results.timeout for users.

I thought the same when I realized these two configs. Actually, I also got confused and spent some time to distinguish them.

And it sounds more like a system error when we exceed `results.timeout`, usually, it means a network issue. Even worse, the external shuffle service could be unresponsive even if we fall back to the original way to fetch the shuffle data.

Given all, shall we make it an `internal` conf? cc @mridulm @Victsm

Reply:
I am fine with making it an internal conf - we can choose to expose it based on how SPARK-33701 evolves.

Reply:
Makes sense. Done.